### ISSUE TITLE
Sapphire 0.6.x + subcall = Indexing block: multiple GasUsedEvent for a transaction

### ISSUE THREAD

**Issue by CedarMist** [oasisprotocol/oasis-web3-gateway#421]
When using the new `subcall` functionality of 0.6, if a GasUsed event is emitted by the subcall, and the contract emits a GasUsed event too.

This causes the JSON-RPC submitted transaction to hang, and all further transactions to hang, this is not good.

```
{"caller":"utils.go:319","gas_used":{"GasUsed":{"amount":29344}},"index":1,"level":"error","module":"indexer","msg":"multiple gas used events for a transaction","prev_used":1000,"ts":"2023-08-28T16:54:59.197057736Z"}
{"caller":"backend.go:138","err":"multiple GasUsedEvent for a transaction","level":"error","module":"indexer","msg":"generateEthBlock failed","ts":"2023-08-28T16:54:59.197102636Z"}
{"caller":"indexer.go:292","err":"indexing block: multiple GasUsedEvent for a transaction","level":"warn","module":"gateway/indexer","msg":"failed to index block","round":36,"runtime_id":"8000000000000000000000000000000000000000000000000000000000000000","ts":"2023-08-28T16:54:59.197118905Z"}
{"caller":"utils.go:319","gas_used":{"GasUsed":{"amount":29344}},"index":1,"level":"error","module":"indexer","msg":"multiple gas used events for a transaction","prev_used":1000,"ts":"2023-08-28T16:55:00.202874544Z"}
{"caller":"backend.go:138","err":"multiple GasUsedEvent for a transaction","level":"error","module":"indexer","msg":"generateEthBlock failed","ts":"2023-08-28T16:55:00.202918732Z"}
```

**Comment by CedarMist** [comment]
Fix implemented in https://github.com/oasisprotocol/oasis-sdk/pull/1465

**Comment by CedarMist** [comment]
This has been fixed with Sapphire 0.6.2-testnet.
